### PR TITLE
[SYCL] Incorporating more feedback from PR

### PR DESCRIPTION
**Pull request by reble**
Addresses: https://github.com/intel/llvm/pull/5626#discussion_r1207019912

Addresses: https://github.com/intel/llvm/pull/5626#discussion_r1207043448

Addresses: https://github.com/intel/llvm/pull/5626#discussion_r1207039331

### CODE REVIEW

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc [hunk 1]
```diff
@@ -624,13 +624,15 @@ Exceptions:
...
 node add(T cgf, const property_list& propList = {});
 ----
-|This function adds a command group function object to a graph. The function
-object statically contains a group of commands, of which a single command is
-executed at runtime. A command group function object may submit any command
-as defined by the core SYCL specification, and any SYCL extensions unless explicitly
-stated otherwise in <<extension-interaction, Interaction With Other Extensions>>.
-The requisites of `cgf` will be used to identify any dependent nodes in the graph
-to form edges with.
+|The `cgf` command group function behaves in much the same was as the command
```
Comment (EwanC) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]:
```suggestion
|The `cgf` command group function behaves in much the same way as the command
```

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc [hunk 2]
```diff
@@ -1201,62 +1144,68 @@ auto node = graph.add([&](sycl::handler& cgh){
...
-user to decide the device each command will be processed for, not the SYCL
-runtime.
+- It is also an error to create a host accessor to a buffer which is used in
+commands which are currently being recorded to a command graph. Attempting to
+construct a host accessor to an incompatible buffer will result in a
+synchronous error being thrown with error code `invalid`.
+
+==== Error Handling
+
+- When a queue is in recording mode asynchronous exceptions will not be
```
Comment (EwanC) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]: This looks slightly odd as a single bullet-point that's a full paragraph, would remove the bullet-point

**Review by EwanC — APPROVED** [review]
LGTM 😃 A couple of superficial single character comments.

File: sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc (hunk 2 above)
Comment (reble) [inline sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc]:
```suggestion
When a queue is in recording mode asynchronous exceptions will not be
```

**Review by Bensuo — APPROVED** [review]

**Pull request merged**